modules/charon-core/src/main/java/org/wso2/charon3/core/config/ExtensionBuilder.java

```diff
@@ -0,0 +1,477 @@
+        }
+
+        public void setDescription(String description) {
+
+            this.description = description;
+        }
+
+        public Boolean getRequired() {
+
+            return required;
+        }
+
+        public void setRequired(Boolean required) {
+
+            this.required = required;
+        }
+
+        public Boolean getCaseExact() {
+
+            return caseExact;
+        }
+
+        public void setCaseExact(Boolean caseExact) {
+
+            this.caseExact = caseExact;
+        }
+
+        public SCIMDefinitions.Mutability getMutability() {
+
+            return mutability;
+        }
+
+        public void setMutability(SCIMDefinitions.Mutability mutability) {
+
+            this.mutability = mutability;
+        }
+
+        public SCIMDefinitions.Returned getReturned() {
+
+            return returned;
+        }
+
+        public void setReturned(SCIMDefinitions.Returned returned) {
+
+            this.returned = returned;
+        }
+
+        public SCIMDefinitions.Uniqueness getUniqueness() {
+
+            return uniqueness;
+        }
+
+        public void setUniqueness(SCIMDefinitions.Uniqueness uniqueness) {
+
+            this.uniqueness = uniqueness;
+        }
+
+        public ArrayList<String> getCanonicalValues() {
+            return canonicalValues;
+        }
+
+
+        public ArrayList<SCIMDefinitions.ReferenceType> getReferenceTypes() {
+            return referenceTypes;
+        }
+
+        public boolean hasChildren() {
+
+            return subAttributes != null ? true : false;
+        }
+
+        public ExtensionAttributeSchemaConfig(Map<String, String> attributeConfig) {
+
+            uri = attributeConfig.get(SCIMConfigConstants.ATTRIBUTE_URI);
+            name = attributeConfig.get(SCIMConfigConstants.ATTRIBUTE_NAME);
+            type = getDefinedDataType(attributeConfig.get(SCIMConfigConstants.DATA_TYPE));
+            multiValued = Boolean.valueOf(attributeConfig.get(SCIMConfigConstants.MULTIVALUED));
+            description = attributeConfig.get(SCIMConfigConstants.DESCRIPTION);
+            required = Boolean.parseBoolean(attributeConfig.get(SCIMConfigConstants.REQUIRED));
+            caseExact = Boolean.parseBoolean(attributeConfig.get(SCIMConfigConstants.CASE_EXACT));
+            mutability = getDefinedMutability(attributeConfig.get(SCIMConfigConstants.MUTABILITY));
+            returned = getDefinedReturned(attributeConfig.get(SCIMConfigConstants.RETURNED));
+            uniqueness = getDefinedUniqueness(attributeConfig.get(SCIMConfigConstants.UNIQUENESS));
+            String subAttributesString = attributeConfig.get(SCIMConfigConstants.SUB_ATTRIBUTES);
+            if (StringUtils.isNotBlank(subAttributesString) && !"null".equalsIgnoreCase(subAttributesString)) {
+                subAttributes = subAttributesString.split(" ");
+            }
+        }
+
+        public ExtensionAttributeSchemaConfig(JSONObject attributeConfigJSON) throws CharonException {
```

Comment:
why do we need this? do we allow to create extension attribute schema using a json file?

Reply:
No. this is a common Extensio Builder. This method will be used by SCIM2UserExtennsioBuilder which takes filepath and json format aatributes